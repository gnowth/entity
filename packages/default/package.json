{
  "author": "Thierry Wong",
  "description": "default",
  "license": "AGPL-3.0-only",
  "main": "src/index.js",
  "module": "src/index.js",
  "name": "@gnowth/default",
  "private": false,
  "repository": "https://github.com/gnowth/entity.git",
  "sideEffects": false,
  "version": "0.5.1",
  "scripts": {
    "build": "rollup -c=../../configs/rollup.config.js",
    "destroy": "rimraf dist",
    "link-index:dist": "json -I -f package.json -e 'this.main=\"dist/index.js\"; this.module=\"dist/index.js\"'",
    "link-index:src": "json -I -f package.json -e 'this.main=\"src/index.js\"; this.module=\"src/index.js\"'",
    "postdestroy": "npm run link-index:src",
    "prebuild": "npm run destroy && npm run link-index:dist"
  },
  "dependencies": {
    "prop-types": "15.7.2"
  },
  "devDependencies": {
    "lodash": "4.17.11",
    "react": "16.8.4"
  },
  "peerDependencies": {
    "lodash": ">=4.12.0",
    "react": ">=16.8.0"
  }
}
