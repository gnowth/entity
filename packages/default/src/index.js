export * from './context';

export { default as useDefault } from './use-default';

export { default as withDefault } from './with-default';
