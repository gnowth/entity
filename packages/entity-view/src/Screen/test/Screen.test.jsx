import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import DuckScreen from '@entity/duck-namespace-screen';
import React from 'react';
import { EntityTitle } from '@entity/core';
import { Query } from '@entity/duck-query';
import { DefaultProvider } from '@gnowth/default';
import { Map } from 'immutable';
import { Provider } from 'react-redux';

import ViewScreen from '..';

const store = configureStore([])(Map({}));

class EntityTest extends EntityTitle {}
EntityTest.duck = new DuckScreen({ app: 'Test', entity: EntityTest });

it('renders correctly', () => {
  const tree = renderer.create(
    <DefaultProvider store={store}>
      <Provider store={store}>
        <ViewScreen
          queryComponentProps={{
            action: () => EntityTest.duck.actions.get(),
            component: () => <div>Mounted</div>,
          }}
          QueryComponent={Query}
        />
      </Provider>
    </DefaultProvider>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
