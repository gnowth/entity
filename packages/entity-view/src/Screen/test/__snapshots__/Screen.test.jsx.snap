// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Card__UICard-sc-12z216h-0 fCrYUk"
/>
`;
